component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/BeamProducerFinder.java

```diff
@@ -79,7 +72,7 @@ Object writeReplace() throws ObjectStreamException {
         return new SerializableService(plugin, ProducerFinder.class.getName());
     }
 
-    static class QueueInput implements Iterator<Record>, Serializable {
+    static class QueueInput extends LifecycleImpl implements Input, Iterator<Record> {
 
```

Comment:
not sure that this comment is applicable, but maybe Sonar find smth, that we didn't see